lightning/src/ln/channelmanager.rs

```diff
@@ -3668,6 +3674,28 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 
 					true
 				});
+
+				channel_state.claimable_htlcs.retain(|payment_hash, htlcs| {
+					if htlcs.is_empty() { return false }
+					if let OnionPayload::Invoice(ref final_hop_data) = htlcs[0].onion_payload {
+						// Check if we've received all the parts we need for an MPP (the value of the parts adds to total_msat).
+						// In this case we're not going to handle any timeouts of the parts here.
+						if final_hop_data.total_msat == htlcs.iter().fold(0, |total, htlc| total + htlc.value) {
+							return true;
+						} else if !htlcs.into_iter().all(|htlc| {
+							htlc.timer_ticks += 1;
+							return htlc.timer_ticks < MPP_TIMEOUT_TICKS
```

Comment:
nit: IMO avoiding double negatives is more readable
so `s/!htlcs/htlcs`, `s/</>=`

Reply:
Yeah, I agree. Looks like I'd also need to change the `.all()` to a `.any()` :)